src/main/resources/jupyter/safe-mode.js

```diff
@@ -0,0 +1,59 @@
+// Adapted from the All of Us Researcher Workbench "Playground Mode"
+// https://github.com/all-of-us/workbench/blob/master/api/cluster-resources/playground-extension.js
+
+// TODO: the following is required until IA-979 is implemented
+// To deploy on a Leonardo cluster:
+// 1. copy to a public GCS location
+// 2. issue a PUT request to <leonardo>/api/cluster/<billing project>/<cluster name>
+//  with a name:path key value pair set in userJupyterExtensionConfig.nbExtensions
+
+// In "Safe Mode", changes are not saved back to GCS. This extension makes
+// minor UI tweaks to differentiate this mode from normal Jupyter usage, and
+// also removes/hides controls relating to persistence. Technically
+// this does not stop autosave from continuing to happen in the background, but
+// the intended use of this plugin is in a separate space from normal operation
+// which does not support localization features.
+
+define([
+    'base/js/namespace'
+], (Jupyter) => {
+  const load = () => {
+    // TODO: query welder for when to enable (IA-979).   always-on, for now
+    const enabled = true;
+    if (!enabled) {
+      return;
+    }
+
+    // Disable UI controls/notifications relating to saving.
```

Comment:
We should disable the "new notebook" menu item and button for now as well. Marty's recommendation is that "create a new notebook" should take you back to the linked source directory; we're not quite ready for that so we can just hide it for now.

Reply:
Next PR does that :)